src/docfx/build/dependency/DependencyResolver.cs

```diff
@@ -326,56 +312,23 @@ private Document TryResolveFile(Document referencingFile, string pathToDocset)
             }
 
             // resolve from fallback docset
-            if (_fallbackDocset != null &&
-                File.Exists(Path.Combine(_fallbackDocset.DocsetPath, pathToDocset)))
-            {
-                return Document.Create(_fallbackDocset, new FilePath(pathToDocset, FileOrigin.Fallback), _templateEngine);
-            }
-
-            return default;
-        }
-
-        private Document TryResolveResourceFromHistory(GitCommitProvider gitCommitProvider, string pathToDocset, TemplateEngine templateEngine)
-        {
-            if (string.IsNullOrEmpty(pathToDocset))
-            {
-                return default;
-            }
-
-            // try to resolve from source repo's git history
-            if (_fallbackDocset != null && Document.GetContentType(pathToDocset) == ContentType.Resource)
+            if (_fallbackDocset != null)
             {
-                var (repo, _, commits) = gitCommitProvider.GetCommitHistory(_fallbackDocset, pathToDocset);
-                if (repo != null && commits.Length > 0)
+                if (File.Exists(Path.Combine(_fallbackDocset.DocsetPath, pathToDocset)))
                 {
-                    return Document.Create(_fallbackDocset, new FilePath(pathToDocset, FileOrigin.Fallback), templateEngine, isFromHistory: true);
+                    return Document.Create(_fallbackDocset, new FilePath(pathToDocset, FileOrigin.Fallback), _templateEngine);
                 }
-            }
 
-            return default;
-        }
-
-        private (string content, Document file) TryResolveContentFromHistory(GitCommitProvider gitCommitProvider, string pathToDocset, TemplateEngine templateEngine)
-        {
-            if (string.IsNullOrEmpty(pathToDocset))
-            {
-                return default;
-            }
-
-            // try to resolve from source repo's git history
-            if (_fallbackDocset != null)
-            {
-                var (repo, pathToRepo, commits) = gitCommitProvider.GetCommitHistory(_fallbackDocset, pathToDocset);
-                if (repo != null)
+                // resolve from fallback docset git commit history
+                if (lookupFallbackCommits)
                 {
-                    var repoPath = PathUtility.NormalizeFolder(repo.Path);
-                    if (commits.Length > 1)
+                    var (repo, _, commits) = _gitCommitProvider.GetCommitHistory(_fallbackDocset, pathToDocset);
+
+                    // TODO: need to get the first commit that contains the data
```

Comment:
Could you elaborate a bit more about this TODO, does this mean `commits[1]` might not be the one containing data?